### PR TITLE
Use the default metricsFactory if not provided

### LINKED ISSUE #738: Jaeger Agent metrics not returned
As per the [documentation](http://jaeger.readthedocs.io/en/latest/monitoring/) the Jaeger Agent should expose some metrics properly formatted for  Prometheus on `:5778/metrics` 

The Agent is started using:
```
docker run \
  --rm \
  -p5775:5775/udp \
  -p6831:6831/udp \
  -p6832:6832/udp \
  -p5778:5778/tcp \
  jaegertracing/jaeger-agent:1.2 \
  /go/bin/agent-linux --collector.host-port=my-collector:some-port
```

And I tried to retrieve the metrics using:
```
 curl -v localhost:5778/metrics
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 5778 (#0)
> GET /metrics HTTP/1.1
> Host: localhost:5778
> User-Agent: curl/7.54.0
> Accept: */*
> 
< HTTP/1.1 400 Bad Request
< Content-Type: text/plain; charset=utf-8
< X-Content-Type-Options: nosniff
< Date: Wed, 14 Mar 2018 15:35:50 GMT
< Content-Length: 42
< 
'service' parameter must be provided once
```

### PR DESCRIPTION
**Pull request by ledor473**
This will fix #738 

I'm looking at adding a test to ensure the functionality will stay over time. I'll update the PR

### CODE REVIEW

**Comment by ledor473** [comment]
A test case was added, but an error is returned when `withRunningAgent` is used twice:

```
panic: duplicate metrics collector registration attempted [recovered]
	panic: duplicate metrics collector registration attempted

goroutine 73 [running]:
testing.tRunner.func1(0xc420294000)
	/usr/local/Cellar/go/1.10/libexec/src/testing/testing.go:742 +0x29d
panic(0x15a33e0, 0xc4202404c0)
	/usr/local/Cellar/go/1.10/libexec/src/runtime/panic.go:505 +0x229
github.com/jaegertracing/jaeger/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc42001ea00, 0xc420258210, 0x1, 0x1)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/vendor/github.com/prometheus/client_golang/prometheus/registry.go:404 +0x9e
github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics/prometheus.(*vectorCache).getOrMakeCounterVec(0xc420244480, 0x0, 0x0, 0x0, 0x0, 0xc420270200, 0x31, 0xc420270200, 0x31, 0x0, ...)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics/prometheus/cache.go:50 +0x213
github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics/prometheus.(*Factory).Counter(0xc420242440, 0xc420270200, 0x31, 0xc4202447e0, 0x0, 0x0)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics/prometheus/factory.go:110 +0x186
github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics.initMetrics(0x1514200, 0xc42026e280, 0x168c720, 0xc420242440, 0x0, 0x50, 0x15c5e40)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics/metrics.go:72 +0x43e
github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics.Init(0x1514200, 0xc42026e280, 0x168c720, 0xc420242440, 0x0)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-lib/metrics/metrics.go:25 +0x57
github.com/jaegertracing/jaeger/cmd/agent/app/reporter/tchannel.New(0x161f664, 0x10, 0xc420298000, 0x0, 0x168c720, 0xc420242300, 0xc42024e600, 0xc420242301)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/reporter/tchannel/reporter.go:78 +0x283
github.com/jaegertracing/jaeger/cmd/agent/app/reporter/tchannel.(*Builder).CreateReporter(0xc4201ecee8, 0x168c720, 0xc420242300, 0xc42024e600, 0xc420242300, 0x0, 0x0)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/reporter/tchannel/builder.go:119 +0x142
github.com/jaegertracing/jaeger/cmd/agent/app.(*Builder).createMainReporter(0xc4201ece90, 0x168c720, 0xc420242300, 0xc42024e600, 0x0, 0xc4202401c0, 0xc420278140)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/builder.go:113 +0x4f
github.com/jaegertracing/jaeger/cmd/agent/app.(*Builder).CreateAgent(0xc420077e90, 0xc42024e600, 0x2, 0x2, 0x0)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/builder.go:129 +0x8b
github.com/jaegertracing/jaeger/cmd/agent/app.withRunningAgent(0xc420294000, 0xc420077f88)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/agent_test.go:101 +0x150
github.com/jaegertracing/jaeger/cmd/agent/app.TestAgentMetricsEndpoint(0xc420294000)
	/Users/ledor473/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/agent_test.go:71 +0x46
```

**Comment by ledor473** [comment]
The test is working now (had to re-create the `prometheus.DefaultRegistry` before each test).  
I've also squash my commits and add the sign-off needed for the DCO

**Comment by black-adder** [comment]
looks like tests fail due to formatting, run `make fmt`

**Comment by ledor473** [comment]
All green now! 👍

File: cmd/agent/app/agent_test.go [hunk 1]
```diff
@@ -51,6 +94,10 @@ func TestAgentStartStop(t *testing.T) {
 		HTTPServer: HTTPServerConfiguration{
 			HostPort: ":0",
 		},
+		Metrics: jmetrics.Builder{
+			Backend:   "prometheus",
+			HTTPRoute: "/metrics",
```
Comment (ledor473) [inline cmd/agent/app/agent_test.go]: I had to place the default values here as they are usually initialized using Viper

File: cmd/agent/app/agent_test.go [hunk 2]
```diff
@@ -37,7 +39,48 @@ func TestAgentStartError(t *testing.T) {
...
+		resp, err := http.Get(url)
+		require.NoError(t, err)
+		body, err := ioutil.ReadAll(resp.Body)
+		assert.NoError(t, err)
+		assert.Contains(t, string(body), "# HELP")
+	})
+}
+
+func withRunningAgent(t *testing.T, testcase func(string, chan error)) {
+	withNewPrometheusRegistry()
```
Comment (yurishkuro) [inline cmd/agent/app/agent_test.go]: nit: resetDefaultPrometheusRegistry

**Comment by yurishkuro** [comment]
Thanks for the fix @ledor473

**Pull request merged**